src/python/pants/util/resources.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+
+import importlib
+from itertools import chain
+
+import importlib_resources as resources
+
+
+def read_resource(package_or_module: str, resource: str) -> bytes:
+    """Reads a resource file from within the Pants package itself.
+
+    This helper function is designed for compatibility with `pkgutil.get_data()` wherever possible,
+    but also allows compability with PEP302 pluggable importers such as included with PyOxidizer.
+    This requires that resources are loaded from a valid Python package (i.e. must have an
+    `__init__.py` file in the directory).
+    """
+
+    a = importlib.import_module(package_or_module)
+    package_ = a.__package__
+
+    if package_ is None:
```

Comment:
Yup, unpackaged modules will have an empty package name, and `importlib_resources` will raise an appropriate error:

```
>>> importlib_resources.read_binary("", "boop.txt")
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/Users/chrisjrn/.pex/installed_wheels/2238159eb743bd85304a16e0536048b3e991c531d1cd51c4a834d1ccf2829057/importlib_resources-5.0.7-py3-none-any.whl/importlib_resources/_py3.py", line 69, in read_binary
    with open_binary(package, resource) as fp:
  File "/Users/chrisjrn/.pex/installed_wheels/2238159eb743bd85304a16e0536048b3e991c531d1cd51c4a834d1ccf2829057/importlib_resources-5.0.7-py3-none-any.whl/importlib_resources/_py3.py", line 24, in open_binary
    package = _common.get_package(package)
  File "/Users/chrisjrn/.pex/installed_wheels/2238159eb743bd85304a16e0536048b3e991c531d1cd51c4a834d1ccf2829057/importlib_resources-5.0.7-py3-none-any.whl/importlib_resources/_common.py", line 66, in get_package
    resolved = resolve(package)
  File "/Users/chrisjrn/.pex/installed_wheels/2238159eb743bd85304a16e0536048b3e991c531d1cd51c4a834d1ccf2829057/importlib_resources-5.0.7-py3-none-any.whl/importlib_resources/_common.py", line 57, in resolve
    return cand if isinstance(cand, types.ModuleType) else importlib.import_module(cand)
  File "/opt/homebrew/Cellar/python@3.9/3.9.13_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1027, in _gcd_import
  File "<frozen importlib._bootstrap>", line 961, in _sanity_check
ValueError: Empty module name
```

Reply:
Why then do you check for `None` and not falsey? What case gives a `__package__` of `None`?

Reply:
the type annotation on `ModuleType.__package__` is `str | None`, so it's just there to placate the type checker with something nicer than an `assert`.

Reply:
Ok. That's likely because it can be `None` in Pythons Pants does not use, definitely in 2.7 anyhow. At any rate, it seems a comment or a straight up # type: ignore[appropriate thing] with comment explaining why leaves the reader in a much better position. I'd naively want to "fix" this only to later get stomped by `./pants check`.

Reply:
Done.